proposals/0048-container-api.md

```diff
@@ -0,0 +1,231 @@
+
+## Public APIs
+
+- `AstroContainer::create`: creates a new instance of the container.
+- `AstroContainer.renderToString`: renders a component and return a string.
+- `AstroContainer.renderToResponse`: renders a component and returns the `Response` emitted by the rendering phase.
+- `AstroContainer.addServerRenderer`: to programmatically store a renderer inside the container. 
+
+### `create` function
+
+A container is class exposed via the `astro/container` specifier. Users **shouldn't** create a container using the `new` instance, but they should use the static function `create`:
+
+```js
+const container = await AstroContainer.create()
+```
+
+The reason why `create` returns a promise is because internally we will validate and parse the user configuration. As a developer, you will be able to pass the same configuration that Astro uses:
+
+```js
+import astroConfig from "../src/astro.config.mjs";
+
+const container = await AstroContainer.create({
+    astroConfig
+})
+```
+
+The function `create` will throw an error if there's a validation error.
+
+> [!NOTE]
+> If you use a TypeScript file for your configuration, you are responsible for loading and transforming it to JavaScript.
+
+
+The `container` binding will expose a `renderToString` that accepts a Astro component and will return a `string`:
+
+```js
+import astroConfig from "../src/astro.config.mjs";
+
+const container = await AstroContainer.create({
+    astroConfig
+})
+
+const content = await container.renderToString(AstroComponent);
+```
+
+#### Options
+
+It will be possible to tweak the container APIs with options and such. The `.create` function will accept the following interface:
+
+```ts
+type AstroContainerOptions = {
+    streaming: boolean;
+    renderers: AddServerRenderer[];
+    astroConfig: AstroUserConfig;
+}
+```
+
+The `astroConfig` object is literally the same object exposed by the `defineConfig`, inside the `astro.config.mjs` file. This very configuration 
+will go under the same schema validation that Astro uses internally. This means that an invalid schema will result in an error of the `create` function.
+
+#### Add a renderer
+
+Adding a renderer can be done manually or automatically. 
+
+The automatic way is meant for static applications, or cases where the container isn't called at runtime. The renderers maintained by the Astro org will expose
+a method called `getContainerRenderer` that will return the correct information that will tell the container **how to import the renderer**. The importing of the renderer is done via a function called `loadRenderers`, exported by a new virtual module called `astro:container`: 
+
+```js
+import { getContainerRenderer as reactRenderer } from "@astrojs/react";
+import { getContainerRenderer as vueRenderer } from "@astrojs/vue";
+import { laodRenderers } from "astro:container";
```

Comment:
I think this import should be `{ loadRenderers }` instead of `{ laodRenderers }`